go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -0,0 +1,304 @@
+			egressFirewall.Namespace, err)
+	}
+	for _, uuid := range uuids {
+		for _, node := range existingNodes.Items {
+			_, stderr, err := util.RunOVNNbctl("remove", "logical_switch",
+				joinSwitch(node.Name), "acls", uuid)
+			if err != nil {
+				klog.Errorf("Remove failed to delete the rule for "+
+					"address_set=%s, stderr: %q (%v)", nsInfo.addressSet.GetIPv4HashName(), stderr, err)
+			}
+		}
+	}
+}
+
+func (ef *egressFirewall) addACLToJoinSwitch(joinSwitch, hashedAddressSetName string) error {
+	for _, rule := range ef.egressRules {
+		var match string
+		var action string
+		if rule.access == egressfirewallapi.EgressFirewallRuleAllow {
+			action = "allow"
+		} else {
+			action = "drop"
+		}
+
+		if len(rule.ports) == 0 {
+			ipAddress, _, err := net.ParseCIDR(rule.to.cidrSelector)
+			if err != nil {
+				return fmt.Errorf("rule.to.cidrSelector %s is not a valid CIDR (%+v)", rule.to.cidrSelector, err)
+			}
+			if ipAddress.To4() != nil {
+				//rule.to.cidrSelector is an ipv4 address
+				match = fmt.Sprintf("match=\"ip4.dst == %s && ip4.src == $%s\"", rule.to.cidrSelector, hashedAddressSetName)
+			} else {
+				//rule.to.cidrSelector is an ipv6 address
+				match = fmt.Sprintf("match=\"ip6.dst == %s && ip6.src == $%s\"", rule.to.cidrSelector, hashedAddressSetName)
+			}
+		} else {
+			ipAddress, _, err := net.ParseCIDR(rule.to.cidrSelector)
+			if err != nil {
+				return fmt.Errorf("rule.to.cidrSelector %s is not a valid CIDR (%+v)", rule.to.cidrSelector, err)
+			}
+			if ipAddress.To4() != nil {
+				//rule.to.cidrSelector is an ipv4 address
+				match = fmt.Sprintf("match=\"ip4.dst == %s && ip4.src == $%s && ( %s )\"", rule.to.cidrSelector, hashedAddressSetName, egressGetL4Match(rule.ports))
+			} else {
+				//rule.to.cidrSelector is an ipv6 address
+				match = fmt.Sprintf("match=\"ip6.dst == %s && ip6.src == $%s && ( %s )\"", rule.to.cidrSelector, hashedAddressSetName, egressGetL4Match(rule.ports))
+			}
+		}
+		uuid, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading",
+			"--columns=_uuid", "find", "ACL", match, "action="+action,
+			fmt.Sprintf("external-ids:egressFirewall=%s", ef.namespace))
+
+		if err != nil {
+			return fmt.Errorf("executing find ACL command, stderr: %q, %+v", stderr, err)
+		}
+		if uuid == "" {
+			_, stderr, err := util.RunOVNNbctl("--id=@acl", "create", "acl",
+				fmt.Sprintf("priority=%d", defaultStartPriority-rule.id),
+				fmt.Sprintf("direction=%s", fromLport), match, "action="+action,
+				fmt.Sprintf("external-ids:egressFirewall=%s", ef.namespace),
+				"--", "add", "logical_switch", joinSwitch,
+				"acls", "@acl")
+			if err != nil {
+				return fmt.Errorf("executing create ACL command, stderr: %q, %+v", stderr, err)
+			}
+		} else {
+			_, stderr, err := util.RunOVNNbctl("add", "logical_switch", joinSwitch, "acls", uuid)
+			if err != nil {
+				return fmt.Errorf("executing create ACL command, stderr: %q, %+v", stderr, err)
```

Comment:
this is not a create ACL command

Reply:
edited, "adding ACL to joinsSwitch %s failed, stderr: %q, %+v"" done